File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    let arr = new Set();
    
    for(let i = 0; i < elements.length; i++){ // 길이가 i인 연속 부분 수열
        for(let j = 0; j < elements.length; j++) { 
            let sum = 0;
            for(let k = j; k < j + i + 1; k++) {
                if(k >= elements.length) {
                    sum += elements[k - elements.length];
                }
                else sum += elements[k];
            } 
            
            arr.add(sum);
        }
    }    
    
    return arr.size
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    count = 0
    s = set()
    for i in range(len(elements)):
        for j in range(len(elements)):
            if j + i >= len(elements):
                rest = (j + i + 1) - len(elements) 
                front = elements[j:]
                back = elements[:rest]
                cur = [*front, *back]
                s.add(sum(cur))
                
            else:
                cur = elements[j:j + i + 1]
                s.add(sum(cur))
                
    return len(s)
                
    